osf/models/files.py

```diff
@@ -399,56 +206,13 @@ def get_file_guids(cls, materialized_path, provider, node):
 
         return guids
 
-    @classmethod
-    def resolve_class(cls, provider, _type=2):
-        """Resolve a provider and type to the appropriate subclass.
-        Usage:
-            >>> FileNode.resolve_class('box', FileNode.ANY)  # BoxFileNode
-            >>> FileNode.resolve_class('dropbox', FileNode.FILE)  # DropboxFile
-        :rtype: Subclass of FileNode
-        """
-        try:
-            return PROVIDER_MAP[provider][int(_type)]
-        except IndexError:
-            raise exceptions.SubclassNotFound('_type must be 0, 1, or 2')
-        except KeyError:
-            raise exceptions.SubclassNotFound(provider)
-
-    @classmethod
-    def _filter(cls, qs=None):
-        """Creates an odm query to limit the scope of whatever search method
-        to the given class.
-        :param qs RawQuery: An odm query or None
-        :rtype: RawQuery or None
-        """
-        # Build a list of all possible constraints leaving None when appropriate
-        # filter(None, ...) removes all falsey values
-        qs = filter(None, (qs,
-                           Q('is_file', 'eq', cls.is_file) if hasattr(cls, 'is_file') else None,
-                           Q('provider', 'eq', cls.provider) if hasattr(cls, 'provider') else None,
-                           ))
-        # If out list is empty return None; there's no filters to be applied
-        if not qs:
-            return None
-        # Use reduce to & together all our queries. equivalent to:
-        # return q1 & q2 ... & qn
-        return functools.reduce(lambda q1, q2: q1 & q2, qs)
-
-    @classmethod
-    def find(cls, qs=None):
-        """A proxy for StoredFileNode.find but applies class based constraints.
-        Wraps The MongoQuerySet in a GenWrapper this overrides the __iter__ of
-        MongoQuerySet to return wrapped objects
-        :rtype: GenWrapper<MongoQuerySet<cls>>
-        """
-        return utils.GenWrapper(StoredFileNode.find(cls._filter(qs)).order_by('id'))
-
-    @classmethod
-    def find_one(cls, qs):
-        """A proxy for StoredFileNode.find_one but applies class based constraints.
-        :rtype: cls
-        """
-        return StoredFileNode.find_one(cls._filter(qs)).wrapped()
+    def to_storage(self):
+        storage = super(BaseFileNode, self).to_storage()
+        if 'trashed' not in self.type.lower():
+            for key in tuple(storage.keys()):
```

Comment:
It does, at least in tests. I put it in because I was getting back keys that don't belong to a GithubFileNode instance.